core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```
@@ -198,12 +190,19 @@ class HadoopRDD[K, V](
     if (inputFormat.isInstanceOf[Configurable]) {
       inputFormat.asInstanceOf[Configurable].setConf(jobConf)
     }
-    val inputSplits = inputFormat.getSplits(jobConf, minPartitions)
-    val array = new Array[Partition](inputSplits.size)
-    for (i <- 0 until inputSplits.size) {
-      array(i) = new HadoopPartition(id, i, inputSplits(i))
+    // SPARK-5068:catch the exception when the path is not exists
+    try {
+      val inputSplits = inputFormat.getSplits(jobConf, minPartitions)
+      val array = new Array[Partition](inputSplits.size)
+      for (i <- 0 until inputSplits.size) {
+        array(i) = new HadoopPartition(id, i, inputSplits(i))
+      }
+      array
+    } catch {
+      case e: InvalidInputException => {
```

I guess throwing exception is the way how `InputFormat` handle the non-existed file. I agree with @srowen , we couldn't just ignore the error here. Probably it's better to leave the current implementation unchanged.